environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -391,9 +396,10 @@ private void editFreeIPA(EnvironmentEditDto editDto, Environment environment) {
     }
 
     private void editProxyIfChanged(EnvironmentEditDto editDto, Environment env) {
-        ProxyConfig newProxyConfig = editDto.getProxyConfig();
-        if (newProxyConfig != null && !newProxyConfig.equals(env.getProxyConfig())) {
-            throw new NotImplementedException("Editing the proxy configuration is not supported yet");
+        if (editDto.getProxyConfig() != null &&
+                (env.getProxyConfig() == null || !Objects.equals(editDto.getProxyConfig().getName(), env.getProxyConfig().getName()))) {
+            ProxyConfig newProxyConfig = proxyConfigService.getByNameForAccountId(editDto.getProxyConfig().getName(), editDto.getAccountId());
+            proxyConfigService.modify(env, newProxyConfig);
```

Comment:
isn't this one pretty slow in requesttime?

Reply:
it will start a flow after validation